### DISCUSSION TITLE
[Bug] instance and topology are incorrectly displayed

### DISCUSSION THREAD

**Discussion by wzqiang1332** [apache/skywalking#12504] [Q&A]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar issues.

### Apache SkyWalking Component

OAP server (apache/skywalking)

### What happened

skywalking version: 10.0.1
storage: banyandb-0.6.1

the same service_name and instance_name shows two records. I checked the attributes and found that the process ID was different between the two times. Would a new instance record be generated for each deployment?
![image](https://github.com/user-attachments/assets/c95147e3-78e4-413f-8dac-4623265dee27)

redis is configured using hostname in the program configuration file, and the result is displayed in the topology diagram, both ip and hostname. the dimp-redis1 ip is 172.17.0.133
![image](https://github.com/user-attachments/assets/9a3da7e9-f1b4-434d-8afd-3af1db1222e3)
![image](https://github.com/user-attachments/assets/9bed4d2d-3b5f-4f79-b572-ac1d4d9d9bf9)

### What you expected to happen

Only one record of the same instance_name is displayed. Only one record of the same redis instance, ip address, or domain name identified in the topology is displayed

### How to reproduce

Use hostname to configure the redis link, start the service, stop the service, and restart it again.

### Are you willing to submit a pull request to fix on your own?

- [ ] Yes I am willing to submit a pull request on my own!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by wu-sheng** [comment]
This could be caused by different plugin interceptor uses different addresses, before or after dns resolver.
